File: Practica4/FiguraGeometrica.py
import math

class FiguraGeometrica:
    # círculo
    def area_circulo(self, radio):
        return math.pi * radio * radio

    # rectángulo
    def area_rectangulo(self, largo, ancho):
        return largo * ancho

    # triángulo rectángulo
    def area_triangulo(self, base, altura):
        return (base * altura) / 2

    # trapecio
    def area_trapecio(self, baseMayor, baseMenor, altura):
        return ((baseMayor + baseMenor) * altura) / 2

    # pentágono
    def area_pentagono(self, lado):
        return (1.0 / 4.0) * math.sqrt(5 * (5 + 2 * math.sqrt(5))) * lado * lado

    # Método principal para simular sobrecarga
    def area(self, *args):
        if len(args) == 1:
            return self.area_circulo(args[0])
        elif len(args) == 2:
            return self.area_rectangulo(args[0], args[1])
        elif len(args) == 3:
            if isinstance(args[2], bool):
                return self.area_triangulo(args[0], args[1])
            else:
                return self.area_trapecio(args[0], args[1], args[2])
        elif len(args) == 1:
            return self.area_pentagono(args[0])
        else:
            return 0

# Ejemplo de uso
if __name__ == "__main__":
    Fg = FiguraGeometrica()

    print("círculo:", Fg.area(5))
    print("rectángulo:", Fg.area(4, 6))
    print("triángulo rectángulo:", Fg.area(3, 4, True))
    print("trapecio:", Fg.area(4, 6, 5))
    print("pentágono:", Fg.area(3))

File: Practica4/FiguraGeometrica.java
public class FiguraGeometrica {
	double area(double radio) {
		return Math.PI*radio*radio;
	}
	double area(double base,double altura) {
		return base*altura;
	}
	double area(double base, float altura) {
		return(base*altura)/2;
	}
	double area(double base1, double base2,double altura) {
		return((base1*base2)*altura)/2;
	}
	public double area(int lado) {
        return (1.0 / 4.0) * Math.sqrt(5 * (5 + 2 * Math.sqrt(5))) * lado * lado;
    }
	
	public static void main(String[] args) {
		FiguraGeometrica f1= new FiguraGeometrica();
		FiguraGeometrica f2 = new FiguraGeometrica();
		FiguraGeometrica f3 = new FiguraGeometrica();
		FiguraGeometrica f4 = new FiguraGeometrica();
		FiguraGeometrica f5 = new FiguraGeometrica();
		System.out.println("Circulo:  "+f1.area(1));
		System.out.println("Rectangulo:  "+f2.area(2, 4));
		System.out.println("Triangulo Rectangulo: "+f3.area(5, 3));
		System.out.println("Trapecio: "+f4.area(3, 4, 6));
		System.out.println("Pentagono: "+f5.area(6));
	
	}
}